### DISCUSSION TITLE
Problema Aggiornamento Docker su RASPBERRY Pi

### DISCUSSION THREAD

**Discussion by nickylo1978** [indomus/forum#1373] [HUB: Home Assistant]
Salve,
non sono molto pratico ma pur seguendo dettagliatamente la guida sull'aggiornamento mi ritrovo con un problema che non mi spiego.
Seguendo la vostra guida per aggiornamento docker su rasperry all'ultima versione:
'
sudo docker stop home-assistant
sudo docker rm home-assistant
sudo docker pull ghcr.io/home-assistant/home-assistant:stable
'
sudo docker run --init -d --name="home-assistant" -e "TZ=Europe/Rome" -v /home/homeassistant/.homeassistant:/config --net=host --restart=always ghcr.io/home-assistant/home-assistant:stable
'
tutto fila liscio, vengono scaricati gli update e installati.

Il problema compare al primo accesso alla concole web dopo l'aggiornamento: ogni configurazione precedente viene resettata e mi ritrovo con la schermata come l'avessi appena installato per la prima volta.

Sicuramente sbaglio qualcosa, ma non capisco cosa.
Grazie

**Comment by snakuzzo** [comment]
Probabilmente sbagli qualcosa nel mount dei volumi.

**Comment by nickylo1978** [comment]
Premetto che non ho approfondito il discorso dei volumi, speravo che seguendo la guida di indomus.it bastasse per fare l'aggiornamento.
Anche perchè nella guida non vengono menzionati questi volumi a cui ti riferisci.

**Reply by snakuzzo** [reply]
Secondo me hai fatto confusione con i volumi.

Nella guida di installazione si fa riferimento alla directory `/home/pi/homeassistant` come vedi sotto..

```
docker run --init -d --name="home-assistant" -e "TZ=Europe/Rome" -v /home/pi/homeassistant:/config --net=host --restart always ghcr.io/home-assistant/home-assistant:stabl
```

In quella di aggiornamento invece `/home/homeassistant/.homeassistant`

```
docker run --init -d --name="home-assistant" -e "TZ=Europe/Rome" -v /home/homeassistant/.homeassistant:/config --net=host --restart=always ghcr.io/home-assistant/home-assistant:stable
```

Devi capire che quelle sono directory che possono chiamarsi pure `/pippo/pluto/paperino`...dipendono da te e da dove hai deciso di piazzare i file di configurazione.

Il consiglio però resta sempre quello di studiare. Se non conoscete l'argomento, ok a fare copia-incolla, ma dovete capire cosa state facendo. Non sperare che le cose funzionino solo perché sono scritte così.

**Comment by nickylo1978** [comment]
Concordo con te sul fatto dello studio e sopratutto sul non dare per scontato quello che c'è scritto.
Dico solo che essendo una guida, io la uso come studio ma se non mi viene detto attenzione alla directory mi risulta difficile intuirlo da solo.
Grazie cmq credo tu abbiamo centrato il problema, il percorso diverso non l'avevo proprio notato !
Proverò e starò più attento ! Grazie ancora

**Reply by snakuzzo** [reply]
Perfetto. Ovviamente non era una critica distruttiva, ma un suggerimento ;)

**Reply by KC-inDomus** [reply]
nick perdonami, ma se dovessimo scrivere "attento a questo, attento a quello" le guide diventerebbero dei poemi (e lo sono già abbastanza). la critica è "prestate più attenzione _critica_ a quel che fate, più che limitarvi al copia incolla". tutto qui. :)
